### ISSUE TITLE
Error: uneven number of nocov comments detected

### ISSUE THREAD

**Issue by DannyBen** [colszowka/simplecov#846]
In versions prior to 0.18 we could add the `# :nocov:` comment at the beginning of files, without providing a closing tag, to ignore the entire file.

In version 0.18 this errors with `uneven number of nocov comments detected` followed by what seems to be an uncaught exception:

```
simplecov-0.18.0/lib/simplecov/source_file.rb:186:in `block in process_skipped_lines': undefined method `-' for nil:NilClass (NoMethodError)
```

Was the ability to ignore the whole file using just a single `# :nocov:` removed, or is this a bug?
I did not spot anything to that effect in the changelog.

**Comment by PragTob** [comment]
@DannyBen :wave: 

Thanks for the bug report! :green_heart: 

It wasn't intentionally removed, what happened was that the way nocov are calculated was changed so that they can also more easily used for branch coverage and that the same data could be reused for nocov.

I didn't know that it was possible to use it like this (makes sense though).

I'm somewhat certain that I can fix it relatively easily.

Thank you!
Tobi

**Comment by DannyBen** [comment]
> I'm somewhat certain that I can fix it relatively easily.

Excellent. Although it is not a big deal, I sort of got used to this feature, saying ":nocov: from here until the next :nocov: or EOF"

**Comment by PragTob** [comment]
Released in 0.18.1

Happy testing & measuring!

**Comment by DannyBen** [comment]
> Happy testing & measuring!

Thank you for the super quick turnaround. It was a real pleasure watching how the pros work.
